### ISSUE TITLE
Translate to German

### ISSUE THREAD

**Issue by lewtun** [huggingface/course#43]
Hi there 👋

Let's translate the course to German so that the whole community can benefit from this resource 🌎!

Below are the chapters and files that need translating - let us know here if you'd like to translate any and we'll add your name to the list. Once you're finished, open a pull request and tag this issue by including `#issue-number` in the description, where `issue-number` is the number of this issue.

> 🙋 If you'd like others to help you with the translation, you can also post in our [forums](https://discuss.huggingface.co/c/course/20) or tag [@_lewtun](https://twitter.com/_lewtun) on Twitter to gain some visibility.

## Chapters

**0 - Setup**
- [x] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter0/1.mdx) [@MarcusFra](https://github.com/MarcusFra)

**1 - Transformer models**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/1.mdx) [@MarcusFra](https://github.com/MarcusFra)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/8.mdx)
- [ ] [`9.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/9.mdx)
- [ ] [`10.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter1/10.mdx)

**2 - Using 🤗 Transformers**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/1.mdx) [langfab](https://github.com/langfab)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter2/8.mdx)

**3 - Fine-tuning a pretrained model**
- [x] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/1.mdx) [JesperDramsch](https://github.com/JesperDramsch)
- [x] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/2.mdx)
- [x] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/3.mdx)
- [x] [`3_tf.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/3_tf.mdx)
- [x] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/4.mdx)
- [x] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/5.mdx)
- [x] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter3/6.mdx)

**4 - Sharing models and tokenizers**
- [x] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/1.mdx) [fabridamicelli](https://github.com/fabridamicelli)
- [x] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/2.mdx)
- [x] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter4/6.mdx)

**5 - The 🤗 Datasets library**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter5/8.mdx)

**6 -  The 🤗 Tokenizers library**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/3.mdx)
- [ ] [`3b.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/3b.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/8.mdx)
- [ ] [`9.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/9.mdx)
- [ ] [`10.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter6/10.mdx)

**7 - Main NLP tasks**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/4.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/7.mdx)
- [ ] [`8.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/8.mdx)
- [ ] [`9.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter7/9.mdx)

**8 - How to ask for help**
- [ ] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/1.mdx)
- [ ] [`2.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/2.mdx)
- [ ] [`3.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/3.mdx)
- [ ] [`4.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/4.mdx)
- [ ] [`4_tf.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/4_tf.mdx)
- [ ] [`5.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/5.mdx)
- [ ] [`6.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/6.mdx)
- [ ] [`7.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/chapter8/7.mdx)

**Events**
- [x] [`1.mdx`](https://github.com/huggingface/course/blob/main/chapters/en/event/1.mdx)

**Comment by MarcusFra** [comment]
If it's not too urgent, I would translate chapters 0 and 1 for a start. Do you have any timeline in mind?

**Comment by langfab** [comment]
I can continue with translating chapter 2 to German

**Comment by JesperDramsch** [comment]
Happy to give chapter 3 a shot. Any guidelines I can follow?

**Comment by JesperDramsch** [comment]
Oh, common question I run into on German translations. Do we use the formal or informal you? I assume informal but I figured I'd ask.

**Comment by fabridamicelli** [comment]
Hi @lewtun, 
I could give Ch 4 a try!

**Comment by MarcusFra** [comment]
Hey @langfab, @JesperDramsch and @fabridamicelli! I am glad that you also want to work on the German translation!
I've just been two weeks off - I am going to deliver chapters 0 and 1 by likely the end of next week, so the redirections will work then.

@JesperDramsch: Good question whether to use the formal or informal you. I would have gone for the formal style since it's used mostly in German programming books/translations. But I have no preference at all, maybe for this kind of translation the informal form of address is more appropriate. 

I think we still need to agree on the guidelines, like which words to keep in plain english, when terms are translated into german if we add the english term in brackets, e. g. (engl. TermX), and so on. Any preferences on this? For now I would finish chapter 0 and 1 and then harmonize/agree on things we translated differently.

**Comment by JesperDramsch** [comment]
For now, I was going with the informal you, because the course is written quite colloquially, so it would be really weird mix. I'd have some stuff that translates to:

> "Gut gemacht! Im folgenden können Sie..."

As for the translation of terms, I mostly go with "are the English terms used in German" and refer to Wikipedia and Google searches how others use it for backup. So I'm going with anglicisms like "Inputs, Training, Padding, Batch" but translate samples to "Elemente" etc. I think it's a good idea to have the English translation for key terms when you translate them, it's something that my university professors used to do a lot, because in the end you'll need to know the English terms if you work in machine learning regardless.

I wouldn't be opposed to a running list of "canonical translations and anglicisms", but the question is where to keep it honestly.

**Comment by lewtun** [comment]
Hi everyone, I'd also be in favour with using informal German as the text is written in a rather playful style and we might lose that in translation. 

Regarding where to keep a canonical list of translations & general guidelines, I suggest creating a `TRANSLATING.txt` file with the guidelines and a new `glossary` chapter with information that might be useful for German readers. You can see and example of this being done for Persian here: https://github.com/huggingface/course/pull/129/files

**Comment by JesperDramsch** [comment]
Ok awesome. I got started on the glossary and style guide. Happy to take edits on the PR if anyone thinks I made mistakes or my style suggestions go in the wrong direction. And of course additions as we go along.

**Comment by JesperDramsch** [comment]
Ok, chapter 3 is good to go from my side.

**Comment by fabridamicelli** [comment]
@lewtun the subchapters 4.1, 4.2, 4.3 can be marked as done (already merged). I'll keep working on the second part (subchapters 4.4, 4.5, 4.6) and open a new PR when that's ready!

**Comment by tognaeg** [comment]
Hi guys, I could take a crack at chapter 5!

**Comment by fabridamicelli** [comment]
@lewtun the PR #640 concludes chapter 4. So now we can mark subsections 4.4, 4.5 and 4.6 as done as well! :rocket:

**Comment by aegis301** [comment]
Hey guys, is this issue still active? I'd be happy to contribute!
